### ISSUE TITLE
Sync XMLHTTPRequests deprecated by chrome

### ISSUE THREAD

**Issue by sou-chon** [psychopy/psychojs#89]
Reference:
[https://www.chromestatus.com/feature/4664843055398912](https://www.chromestatus.com/feature/4664843055398912)

In Chrome, if users close the experiment before it finishes, the user will get an error of type "sync XHR in page dismissal".
Basically the server doesn't get the DELETE request from the client, although I don't know what effect this has - not closing an experiment properly.

https://github.com/psychopy/psychojs/blob/092829f325a70791685bf648c538ea0740f874e1/js/core/ServerManager.js#L193-L201

**Comment by apitiot** [comment]
This is most vexing, but alas, that was always going to happen.
Thank you for the detailed comment! I shall look into it this week.
Best wishes,

Alain

**Comment by apitiot** [comment]
I am testing a solution, using the Beacon API. So far so good.
I would have preferred fetch/keepalive but there is a CORS bug in Chrome. Nothing is ever easy :-)

**Comment by sou-chon** [comment]
I think Beacon API is a good solution too as it's chrome's migration recommendation.

I just wonder whether it's 100% reliable because I saw this [one article](https://volument.com/blog/sendbeacon-is-broken) saying it's not:
According to the article, "about 30% of browsers that claim to support the beacon API failed to deliver the data to our servers when the page was closed, which is the whole purpose of the sendBeacon call. Linux fails 94.1% of the time". 

But again that's just one article and there's no alternative (that i know of).

And despite the API saying the network request is still sent after the page is navigated away or the tab is closed, I cannot find documentation about how it behaves if the browser is closed altogether; in that case I don't know what it could work since the process is terminated.

I think the foolproof way is to use Beacon API + making the server tolerate not closing the experiment properly (maybe it already does? I haven't seen the code for pavlovia server) like closing it automatically after a timeout. Moreover I don't know the purpose of this network request so I don't know the effect of not receiving it. But i doubt it would affect data collection?  :-)
